### DISCUSSION TITLE
Volumes: Which directories should be persistent?

### DISCUSSION THREAD

**Discussion by julianklotz** [craftcms/docker#64] [Q&A] [answered]
I’m currently setting up a docker-based deployment for CraftCMS. I was wondering which directories should be persistent, i.e. be mapped to volumes? The base images don’t set volumes in the docker files at all.

Instinctively, I’d set `/app/storage` and the (installation-specific) assets folder `/app/web/assets` to be persistent.

**EDIT:** I’ve set craft *not* to allow updates and *not* to allow admin changes in production, so I expect that no config changes are made during runtime.

Any advice from your side?

Best regrads!

**Comment by jawys** [comment, marked as answer]
Hi @julianklotz!

In addition to the mounts you mentioned, I also mount `/app/storage/backups` so another service can process them via a shared mount.
You can also mount a Docker secret for the `license.key` if you run your Docker host in Swarm Mode, so you can manage your license outside your repo.

Best regards!

**Reply by julianklotz** [reply]
Lovely, thank you!
